### ISSUE TITLE
PromptDialog.Choice and FormFlow giving issues while used in Conversation update section.

### ISSUE THREAD

**Issue by Tushti02** [Microsoft/BotFramework-Emulator#342]
[issue_code.txt](https://github.com/Microsoft/BotFramework-Emulator/files/1466428/issue_code.txt)
The prompt dialog appears twice in welcome msg as in this code.
If I use form flow with a class only containing a field named "Start" of enum('yes', 'no as options)  type, it automatically takes a "" (blank string) as input and gives msg ==>' “” is not a start option.' (this happens even before any user input is taken).

(please refer attached txt file for some code section )

Basically all I want to achieve is show a welcome msg as soon as user opens window for chat.
That welcome msg should introduce the bot and ask user if he is ready to begin using a prompt dialog or form flow (or if any other option is possible , please let me know).This I am able to achieve ,with the above mentioned issues though - which do not look good to client :(

**Comment by eanders-MS** [comment]
When a conversation starts, the emulator sends two conversationUpdate messages to the bot: one for the user joining, and another for the bot joining. We structured it this way in order to simulate group conversation, where users come and go during the conversation's lifetime. https://github.com/Microsoft/BotFramework-Emulator/issues/98#issuecomment-277043838 illustrates how to filter out the bot-joined message from the user-joined message. I could be wrong, but I suspect your logic is inverted in this code snippet within your StartAsync method:
```cs
if (newMember.Id == context.Activity.Recipient.Id)
{
    await SendWelcomeMessageAsync(context);                               
}
```
You could try inverting the conditional:
```cs
if (newMember.Id != context.Activity.Recipient.Id)
```
This should filter the bot from the MembersAdded list.

**Comment by Tushti02** [comment]
Yes that change of logic helped but i had to rewrite the logic of OnCategorySelected  function of prompt dialog in Message received async method's else block as well , then it worked.

public virtual async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
        {
            var message = await result;
            if (context.Activity.Type == ActivityTypes.ConversationUpdate)
            {
                IConversationUpdateActivity update = (IConversationUpdateActivity)context.Activity;
                using (var scope = Microsoft.Bot.Builder.Dialogs.Internals.DialogModule.BeginLifetimeScope(Conversation.Container, (IMessageActivity)context.Activity))
                {
                    if (update.MembersAdded.Any())
                    {
                        foreach (var newMember in update.MembersAdded)
                        {
                            if (newMember.Id != context.Activity.Recipient.Id)
                            {
                                await SendWelcomeMessageAsync(context);                               
                            }
                        }
                    }
else
{
//OnCategorySelected methods code
}
                }
            }
}

**Comment by eanders-MS** [comment]
This issue appears resolved. Let me know otherwise and I'll reopen it.
